R/service-navigation.R

```diff
@@ -0,0 +1,138 @@
+#' Service navigation
+#'
+#' @description
+#' Service navigation component consistent with the
+#' [GDS service navigation](https://design-system.service.gov.uk/components/service-navigation/)
+#' (pre-June 2025).
+#'
+#' @param links A vector of actionLinks to be added to the service navigation.
+#' inputIDs are auto-generated and are the snake case version of the link text, e.g.
+#' "Overview page" will have an inputID of overview_page. Can also be provided as a named
+#' vector, in which case the vector names will be used as the page titles and the vector values
+#' will be used as the individual inputIDs.
+#'
+#' @returns Shiny tag object
+#' @export
+#'
+#' @examples
+#' if (interactive()) {
+#'   ui <- shiny::fluidPage(
+#'     shinyGovstyle::header("Title", "Secondary heading"),
+#'     shinyGovstyle::service_navigation(
+#'        c("Summary data", "Detailed stats 1", "User guide")
+#'     ),
+#'     bslib::navset_hidden(
+#'        id = "main_panels",
+#'        bslib::nav_panel(
+#'          "summary_data",
+#'          shiny::tags$h2("Summary data")
+#'        ),
+#'        bslib::nav_panel(
+#'          "detailed_stats_1",
+#'          shiny::tags$h2("Detailed stats 1")
+#'        ),
+#'        bslib::nav_panel(
+#'          "user_guide",
+#'          shiny::tags$h2("User guide")
+#'        )
+#'      ),
+#'      shinyGovstyle::footer(full = TRUE)
+#'   )
+#'
+#'   server <- function(input, output, session) {
+#'       observeEvent(input$summary_data, bslib::nav_select("main_panels", "summary_data"))
+#'       observeEvent(input$detailed_stats_1, bslib::nav_select("main_panels", "detailed_stats_1"))
+#'       observeEvent(input$user_guide, bslib::nav_select("main_panels", "user_guide"))
+#'   }
+#'
+#'   shiny::shinyApp(ui = ui, server = server)
+#' }
+service_navigation <- function(
+  links
+) {
+  if (is.null(names(links))) {
+    link_names <- links
+  } else {
+    link_names <- names(links)
+    link_names[link_names == ""] <- links[link_names == ""]
+  }
+  navigation <- shiny::tags$div(
+    class = "govuk-service-navigation",
+    `data-module` = "govuk-service-navigation",
+    shiny::tags$div(
+      class = "govuk-width-container",
+      shiny::tags$div(
+        class = "govuk-service-navigation__container",
+        shiny::tags$nav(
+          `aria-label` = "Menu",
+          class = "govuk-service-navigation__wrapper",
+          shiny::tags$button(
+            type = "button",
+            class = "govuk-service-navigation__toggle govuk-js-service-navigation-toggle",
+            `aria-controls` = "navigation",
+            hidden = TRUE,
+            "Menu"
+          ),
+          shiny::tags$ul(
```

Comment:
Looks like we're missing the 'active' state somewhere, where the bold line comes underneath to show which page / tab you're on?

<img width="758" height="82" alt="Image" src="https://github.com/user-attachments/assets/9aec3562-21e7-4a01-b163-6c526789d4e8" />